docs/networking/stun.md

```diff
@@ -122,3 +122,48 @@ originate. Using these internal addresses is much more likely to result in a
 successful direct connection.
 
 ![Diagram of a workspace agent and client over VPN](../images/networking/stun3.png)
+
+## Hard NAT
+
+Some NATs are known to use a different port when forwarding requests to the STUN
+server and when forwarding probe packets to peers. In that case, the address a
+peer discovers over the STUN protocol will have the correct IP address, but the
+wrong port. Tailscale refers to this as "hard" NAT in
+[How NAT traversal works (tailscale.com)](https://tailscale.com/blog/how-nat-traversal-works).
+
+It can block direct connections if the other peer is also behind a "hard" NAT or
+is running a firewall (including Windows Defender Firewall). Peers fallback to
+DERP connections if they cannot establish a direct connection.
```

Comment:
> running a firewall that has similar behavior

The issue isn't a combination NAT/firewall where the NAT component is "hard", the issue is that if you're behind a regular firewall and trying to connect to a peer behind a hard NAT, the UDP packets you send out are for a different port than the packets the peer sends you, so the firewall blocks them.

Reply:
My goal wasn't to change the meaning of the sentence but was to make it easier to read, I don't mind if you change my suggestion. I personally think starting with "It can block direct connections" isn't great though.

Reply:
Also it didn't make it clear that direct connections might still work in this scenario either

Reply:
Direct connections can still work with hard NAT to hard NAT with no firewall interfering with it. The paragraph should make that clear rather than saying that direct connections will be blocked.

I don't mind if you change my wording a bit to make it more accurate